lib/iris/tests/experimental/regrid/test_regrid_conservative_via_esmpy.py

```diff
@@ -364,14 +363,14 @@ def test_global_collapse(self):
         c2.add_dim_coord(x_coord_2, 1)
 
         # NOTE: at present, this causes an error inside ESMF ...
-        context = self.assertRaises(ValueError)
+        context = pytest.raises(ValueError)
```

Comment:
More [`PT011`](https://docs.astral.sh/ruff/rules/pytest-raises-too-broad/)

Reply:
This file raised an issue where ESMF doesn't exist, so I had to change the test to `import esmpy as ESMF', both here in the tests and in the actual code. 

Fixed in https://github.com/SciTools/iris/pull/6599/commits/ce8039e7ffe94206694fe05ea854ea2a5c988e4b, but not convinced the regex patterns are strict enough.